pkg/manifests/manifests.go

```diff
@@ -460,6 +460,10 @@ func (f *Factory) AlertmanagerUserWorkload(trustedCABundleCM *v1.ConfigMap) (*mo
 		a.Spec.Resources = *alertmanagerConfig.Resources
 	}
 
+	if alertmanagerConfig.Secrets != nil {
+		a.Spec.Secrets = append(a.Spec.Secrets, alertmanagerConfig.Secrets...)
+	}
```

Comment:
I think that we can skip checking if `alertmanagerConfig.Secrets` is nil.
```suggestion
	a.Spec.Secrets = append(a.Spec.Secrets, alertmanagerConfig.Secrets...)
```

Reply:
+1 though if this check is dropped, the same should be added to `AlertmanagerMain()` at  https://github.com/openshift/cluster-monitoring-operator/blob/d4d3ad1812c797080befa27592717e0c53ecf81a/pkg/manifests/manifests.go#L585